cli/src/main/java/com/devonfw/tools/ide/tool/lazydocker/LazyDocker.java

```diff
@@ -0,0 +1,70 @@
+package com.devonfw.tools.ide.tool.lazydocker;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.process.ProcessErrorHandling;
+import com.devonfw.tools.ide.process.ProcessMode;
+import com.devonfw.tools.ide.process.ProcessResult;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.util.Set;
+
+/**
+ * {@link ToolCommandlet} for <a href="https://github.com/jesseduffield/lazydocker">lazydocker</a>.
+ */
+public class LazyDocker extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public LazyDocker(IdeContext context) {
+
+    super(context, "lazydocker", Set.of(Tag.DOCKER));
+  }
+
+  @Override
+  public boolean install(boolean silent) {
+
+    String bashPath = this.context.findBash();
+    String command = "docker version --format '{{.Client.APIVersion}}'";
+    ProcessContext pc = this.context.newProcess().errorHandling(ProcessErrorHandling.NONE).executable(bashPath)
+        .addArgs("-c", command);
+    ProcessResult result = pc.run(ProcessMode.DEFAULT_CAPTURE);
+
+    if (result.getOut().isEmpty()) {
+      this.context.info("Docker is not installed, but required for lazydocker.");
+      this.context.info("To install docker, call the following command:");
+      this.context.info("ide install docker");
+      return false;
+    }
+
+    VersionIdentifier dockerAPIversion = VersionIdentifier.of(result.getOut().get(0).toString());
+
+    if (dockerAPIversion.compareVersion(VersionIdentifier.of("1.25")).isLess()) {
+      this.context.info("The installed version of docker does not meet the requirements of lazydocker.");
+      this.context.info("Please upgrade your installation of docker, before installing lazydocker.");
+      return false;
+    }
+
+    command = "docker-compose version --short";
+    pc = this.context.newProcess().errorHandling(ProcessErrorHandling.NONE).executable(bashPath).addArgs("-c", command);
+    result = pc.run(ProcessMode.DEFAULT_CAPTURE);
+
+    if (!result.getOut().isEmpty()) {
+      VersionIdentifier dockercomposeversion = VersionIdentifier.of(result.getOut().get(0).toString());
+
+      if (dockercomposeversion.compareVersion(VersionIdentifier.of("1.23.2")).isLess()) {
+        this.context.info("Found docker-compose version:" + dockercomposeversion);
+        this.context.info(
+            "If you want to use docker-compose with lazydocker, then you'll need at least version 1.23.2 of docker-compose");
+      }
+    }
```

Comment:
should we invent a `preInstall` method like the `postInstall`?

Reply:
Another problem I see with the current approach is that `install` method is called to ensure that a tool is installed even if it is already installed. In case we would have a transitive dependency you would spam the logs with all these messages about lazydocker even if it is already installed for the installation of another tool what could cause confusion. Maybe not a very significant scenario but from that PoV we could even decide to remove this logic at all since docker `< 1.25` is not really relevant anymore.
Another side-node:
If docker is running, I get this:
```
$ docker version --format '{{.Client.APIVersion}}'
1.44
```
But if docker is not running, I get this:
```
$ docker version --format '{{.Client.APIVersion}}'
1.45
error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.45/version": open //./pipe/docker_engine: The system cannot find the file specified.
```
I would rather consider all this a design flaw of docker, but I am wondering how this IDEasy could would behave here. At least to my tests it seems that the error message is written to standard error and looking at our code, it would be ignored just like the rest of the result like the return code (which is `1` if docker daemon is not running).

Reply:
I think there is no irrelevant log spam if `install lazydocker` would be called when it is already installed imho. It would only produce info's if docker had been removed after installing lazydocker, which I think is very relevant, since it renders lazydocker useless. The warning about docker-compose being out of date also seems relevant to me when checking the lazydocker installation.  

I intentionally ignored the return code and the error message of the version check, since if docker is running or not does not affect the installation of lazydocker. Once lazydocker is installed and started, it will itself notify the user about docker not running. 

But please let me know if you preferred the version checks to be removed. If not, I will be moving them to a `preInstall `method.